### PR TITLE
fix(ci): publish linux and mac artifacts in tar.gz format

### PR DESCRIPTION
**Pull request by nishantwrp**
Signed-off-by: Nishant Mittal <mittalnishant14@outlook.com>

**Description**

This PR fixes part of #5881 

**Notes for Reviewers**
Publish linux & macos artifacts for mesheryctl in `tar.gz` format. Artifacts for windows will still be published in `zip` format.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ builds:
 
   env:
     - CGO_ENABLED=0
-    
+
```
Comment (hexxdump) [inline .goreleaser.yml:17]: remove the extra lines, lint is complaining about these.

File: .goreleaser.yml [hunk 2]
```diff
@@ -50,10 +50,10 @@ archives:
     windows: Windows
     386: i386
     amd64: x86_64
-  format: zip
-#   format_overrides:
-#     - goos: windows
-#       format: zip
+  format: tar.gz
```
Comment (hexxdump) [inline .goreleaser.yml:53]: Looks good.

**Review by hexxdump — COMMENTED** [review]
changes look good, pls fix the lint errors

File: .goreleaser.yml (hunk 1 above)
Comment (nishantwrp) [reply .goreleaser.yml:17]: Actually, these are not newlines. My code editor automatically removed some trailing space these lines had.

On running lint checks locally, looks like the lint is complaining about some other files.

```
➜ golangci-lint run      
internal/graphql/resolver/addons.go:58: unnecessary trailing newline (whitespace)

	}()
internal/graphql/resolver/addons.go:56: unnecessary trailing newline (whitespace)

		}
internal/graphql/resolver/control-plane.go:56: unnecessary trailing newline (whitespace)

			}(ctxID)
```

PS: I don't think golangci-lint even checks `yaml` files.

Comment (nishantwrp) [reply .goreleaser.yml:17]: Made a pull request to fix these problems: https://github.com/meshery/meshery/pull/5890

**Review by hexxdump — APPROVED** [review]
LGTM

**Comment by hexxdump** [comment]
Lets wait for the meshery bash script changes to be ready, before we merge this.

**Review by AugustasV — APPROVED** [review]
LGTM

**Pull request merged**